oauth2client/service_account.py

```diff
@@ -461,3 +462,56 @@ def create_delegated(self, sub):
         result._private_key_pkcs12 = self._private_key_pkcs12
         result._private_key_password = self._private_key_password
         return result
+
+
+class SelfSignedJWTCredentials(object):
+    """Self signed JWT credentials.
+
+    Makes an assertion to server using a self signed JWT from service account
+    credentials.  These credentials do NOT use OAuth 2.0 and instead authenticate
+    directly.
+    """
+    MAX_TOKEN_LIFETIME_SECS = 3600
+    """Default lifetime of the token (one hour, in seconds)."""
+
+    def __init__(self, creds, audience, time_limit):
+        self._creds = creds
+        self._audience = audience
+        self._time_limit = time_limit
+    
+    def get_access_token(self):
+        """Create a signed jwt
+            Returns:
+                An AccessTokenInfo with the signed jwt
+        """
+        now = int(time.time())
+        payload = {
+            'aud': self._audience,
+            'iat': now,
+            'exp': now + self._time_limit,
+            'iss': self._creds._service_account_email,
+            'sub': self._creds._service_account_email
```

Comment:
Delegated credentials or using the envelope pattern for asserting a user across requests, for example:

```python
    payload = {
        'iat': now,
        'exp': now + credentials.MAX_TOKEN_LIFETIME_SECS,
        # aud must match 'audience' in the security configuration in your
        # open api spec. It can be any string.
        'aud': 'echo.esp.sample.cloud.google.com',
        # iss must match 'issuer' in the security configuration in your
        # open api spec. it is typically the service account email, but
        # may be any string that uniquely identifies the issuer.
        'iss': service_account_email,
        # sub and email are mapped to the user id and email respectively.
        'sub': '12345678',
        'email': 'user@example.com'
    }
```